pkgs/by-name/go/gossip/package.nix

```diff
@@ -0,0 +1,97 @@
+{ cmake
+, darwin
+, fetchFromGitHub
+, ffmpeg
+, fontconfig
+, git
+, lib
+, libGL
+, libxkbcommon
+, openssl
+, pkg-config
+, rustPlatform
+, stdenv
+, wayland
+, xorg
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "gossip";
+  version = "0.9";
+
+  src = fetchFromGitHub {
+    hash = "sha256-m0bIpalH12GK7ORcIk+UXwmBORMKXN5AUtdEogtkTRM=";
+    owner = "mikedilger";
+    repo = "gossip";
+    rev = version;
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "ecolor-0.23.0" = "sha256-Jg1oqxt6YNRbkoKqJoQ4uMhO9ncLUK18BGG0fa+7Bow=";
+      "egui-video-0.1.0" = "sha256-3483FErslfafCDVYx5qD6+amSkfVenMGMlEpPDnTT1M=";
+      "ffmpeg-next-6.0.0" = "sha256-EkzwR5alMjAubskPDGMP95YqB0CaC/HsKiGVRpKqUOE=";
+      "ffmpeg-sys-next-6.0.1" = "sha256-UiVKhrgVkROc25VSawxQymaJ0bAZ/dL0xMQErsP4KUU=";
+      "gossip-relay-picker-0.2.0-unstable" = "sha256-3rbjtpxNN168Al/5TM0caRLRd5mxLZun/qVhsGwS7wY=";
+      "heed-0.20.0-alpha.6" = "sha256-TFUC6SXNzNXfX18/RHFx7fq7O2le+wKcQl69Uv+CQkY=";
+      "nip44-0.1.0" = "sha256-of1bG7JuSdR19nXVTggHRUnyVDMlUrkqioyN237o3Oo=";
+      "nostr-types-0.7.0-unstable" = "sha256-B+hOZ4TRDSWgzyAc/yPiTWeU0fFCBPJG1XOHyoXfuQc=";
+      "qrcode-0.12.0" = "sha256-onnoQuMf7faDa9wTGWo688xWbmujgE9RraBialyhyPw=";
+      "sdl2-0.35.2" = "sha256-qPId64Y6UkVkZJ+8xK645at5fv3mFcYaiInb0rGUfL4=";
+      "speedy-0.8.6" = "sha256-ltJQud1kEYkw7L2sZgPnD/teeXl2+FKgyX9kk2IC2Xg=";
+    };
+  };
+
+  # See https://github.com/mikedilger/gossip/blob/0.9/README.md.
+
+  # Compile binary optimized for exact processor with the newest features enabled.
+  RUSTFLAGS = "-C target-cpu=native --cfg tokio_unstable";
```

Comment:
Not sure this optimization should be made, I'd assume it could lead to issues when the package is built on a system with newer cpu features and then substituted to a system that hasn't got these optimizations? But I'm not a rust expert...